aptos-move/aptos-vm/src/sharded_block_executor/cross_shard_state_view.rs

```diff
@@ -179,3 +180,55 @@ mod tests {
         wait_thread.join().unwrap();
     }
 }
+
+pub const TOTAL_SUPPLY_AGGR_BASE_VAL: u128 = u128::MAX >> 1;
+#[derive(Clone)]
+pub struct CrossShardStateViewAggrOverride<'a, S> {
+    cross_shard_state_view: CrossShardStateView<'a, S>,
+}
+
+impl<'a, S: StateView + Sync + Send> CrossShardStateViewAggrOverride<'a, S> {
+    pub fn new(cross_shard_keys: HashSet<StateKey>, base_view: &'a S) -> Self {
+        Self {
+            cross_shard_state_view: CrossShardStateView::new(cross_shard_keys, base_view),
+        }
+    }
+
+    pub fn set_value(&self, state_key: &StateKey, state_value: Option<StateValue>) {
+        self.cross_shard_state_view
```

Comment:
This really belongs in `CrossShardStateView` instead. See my comment above for making this a generic StateView that doesn't care about the `CrossShardStateView`

Reply:
done